.golangci.yml

```diff
@@ -1,80 +1,110 @@
     - wrapcheck         # don't _always_ need to wrap errors
     - wsl               # over-generous whitespace violates house style
-issues:
-  exclude-use-default: true
-  exclude-rules:
-    # Loosen requirements on conformance executor
-    - path: internal/cmd/
-      linters:
-        - forbidigo
-        - errorlint
-    # Loosen requirements on tests
-    - path: _test.go
-      linters:
-        - funlen
-        - varnamelen
-        - unparam
-        - gosec
-    - path: cel/library.go
-      linters:
-        # setting up custom functions/overloads appears duplicative (false positive)
-        - dupl
-        # Types are checked internally within CEL. There are bigger issues if its
-        # type analysis is wrong
-        - forcetypeassert
-    # static unexported lookup tables
-    - path: lookups.go
-      linters:
-        - gochecknoglobals
-    - path: resolver/resolver.go
-      linters:
-        # uses deprecated fields on protoimpl.ExtensionInfo but its the only way
-        - staticcheck
-    # We allow a global validator.
-    - path: validator.go
-      linters:
-        - gochecknoglobals
+  settings:
+    errcheck:
+      check-type-assertions: true
+    forbidigo:
+      forbid:
+        - pattern: ^fmt\.Print
+        - pattern: ^log\.
+        - pattern: ^print$
+        - pattern: ^println$
+        - pattern: ^panic$
+    godox:
+      # TODO, OPT, etc. comments are fine to commit. Use FIXME comments for
+      # temporary hacks, and use godox to prevent committing them.
+      keywords:
+        - FIXME
+    varnamelen:
+      ignore-decls:
+        - T any
+        - i int
+        - wg sync.WaitGroup
+        - ok bool
+        - w io.Writer
+  exclusions:
+    generated: lax
+    presets:
+      - comments
+      - common-false-positives
+      - legacy
+      - std-error-handling
+    rules:
+      # Loosen requirements on conformance executor
+      - linters:
+          - errorlint
+          - forbidigo
+        path: internal/cmd/
+      # Loosen requirements on tests
+      - linters:
+          - funlen
+          - gosec
+          - gosmopolitan
+          - unparam
+          - varnamelen
+        path: _test.go
+      - linters:
+          # setting up custom functions/overloads appears duplicative (false positive)
+          - dupl
+          # Types are checked internally within CEL. There are bigger issues if its
+          # type analysis is wrong
+          - forcetypeassert
+        path: cel/library.go
+      # static unexported lookup tables
+      - linters:
+          - gochecknoglobals
+        path: lookups.go
+      - linters:
+          # uses deprecated fields on protoimpl.ExtensionInfo but its the only way
+          - staticcheck
+        path: resolver/resolver.go
+      # We allow a global validator.
+      - linters:
+          - gochecknoglobals
+        path: validator.go
+      ### BEGIN Temporary exclusions from golangci-lint upgrade.
```

Comment:
I'd rather fix these in follow ups as we may want to decide (1) if they are worth enabling, and (2) pay closer attention to code changes.